packages/server/src/api/controllers/row/ExternalRequest.ts

```diff
@@ -248,6 +163,117 @@ export class ExternalRequest<T extends Operation> {
     if (datasource && datasource.entities) {
       this.tables = datasource.entities
     }
+    this.tableList = Object.values(this.tables)
+  }
+
+  private prepareFilters(
+    id: string | undefined | string[],
+    filters: SearchFilters,
+    table: Table
+  ) {
+    const tables = this.tableList
+    // replace any relationship columns initially, table names and relationship column names are acceptable
+    const relationshipColumns = sdk.rows.filters.getRelationshipColumns(table)
+    filters = sdk.rows.filters.updateFilterKeys(
+      filters,
+      relationshipColumns
+        .map(({ name, definition }) => {
+          const { tableName } = breakExternalTableId(definition.tableId)
+          return {
+            original: name,
+            updated: tableName!,
+          }
+        })
+        // don't update table names - include this for context incase a column would be replaced
+        .concat(
+          tables.map(table => {
+            const tableName = table.originalName || table.name
+            return {
+              original: tableName,
+              updated: tableName,
+            }
+          })
+        )
+    )
+    const primary = table.primary
+    // if passed in array need to copy for shifting etc
+    let idCopy: undefined | string | any[] = cloneDeep(id)
+    if (filters) {
+      // need to map over the filters and make sure the _id field isn't present
```

Comment:
I know that you didn't write this, but how come? I can't think of a good reason you'd need to do this.

Reply:
the `_id` field doesn't exist in the external DB, so it needs to be extracted so that it can be handled as a separate case, rather than passed down to the SQL layer to generate, we convert it to a filter based on the primary keys (the later part of this function).